packages/flutter/lib/src/rendering/selection.dart

```diff
@@ -375,26 +375,39 @@ class SelectWordSelectionEvent extends SelectionEvent {
 ///
 /// The [globalPosition] contains the new offset of the edge.
 ///
-/// This event is dispatched when the framework detects [DragStartDetails] in
+/// The [granularity] contains the granularity that the selection edge should move by.
+///
+/// This event is dispatched when the framework detects [TapDragStartDetails] in
 /// [SelectionArea]'s gesture recognizers for mouse devices, or the selection
 /// handles have been dragged to new locations.
 class SelectionEdgeUpdateEvent extends SelectionEvent {
   /// Creates a selection start edge update event.
   ///
   /// The [globalPosition] contains the location of the selection start edge.
+  ///
+  /// The [granularity] contains the granularity which the selection edge should move by.
   const SelectionEdgeUpdateEvent.forStart({
-    required this.globalPosition
-  }) : super._(SelectionEventType.startEdgeUpdate);
+    required this.globalPosition,
+    TextGranularity? textGranularity
+  }) : granularity = textGranularity ?? TextGranularity.character, super._(SelectionEventType.startEdgeUpdate);
```

Comment:
Should these two names be the same? `textGranularity` and `granularity`.

Reply:
Good point!